src/tests/Product.test.js

```diff
@@ -0,0 +1,37 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import Product from '../components/Product';
+import '@testing-library/jest-dom';
+
+describe('<Product />', () => {
+  it('renders the product name', () => {
+    const testProduct = { name: 'Blueberry', description: 'Ready for you to make the best muffins.' };
+    const testAddToCart = jest.fn();
+
+    render(<Product product={testProduct} addToCart={testAddToCart}/>);
+    const productName = screen.getByText(testProduct.name);
+
+    expect(productName).toBeInTheDocument();
+  });
+
+  it('renders the product description', () => {
+    const testProduct = { name: 'Blueberry', description: 'Ready for you to make the best muffins.' };
+    const testAddToCart = jest.fn();
+
+    render(<Product product={testProduct} addToCart={testAddToCart}/>);
+    const productDescription = screen.getByText(testProduct.description);
+
+    expect(productDescription).toBeInTheDocument();
+  });
+
+  it('calls addToCart function once', () => {
+    const testProduct = { name: 'Blueberry', description: 'Ready for you to make the best muffins.' };
+    const testAddToCart = jest.fn();
+
+    render(<Product product={testProduct} addToCart={testAddToCart}/>);
+    const buttonAddToCart = screen.getByRole('button', {name: 'Add to cart'});
+    userEvent.click(buttonAddToCart);
+
+    expect(testAddToCart).toHaveBeenCalledTimes(1);
+  });
```

Comment:
Nos poderiamos adicionar aqui um describe para dizer que esse it ocorre quando o botão é clicado.
Poderiamos tb usar o .toHaveBeenNthCalledWith` https://jestjs.io/pt-BR/docs/expect#tohavebeennthcalledwithnthcall-arg1-arg2-